$NetBSD: patch-as,v 1.20 2015/02/15 02:11:03 ryoon Exp $

* Fix _res is not supported for multi-threaded programs. error.
* Treat DragonFly like FreeBSD.

--- mozilla/js/src/configure.in.orig	2015-02-05 04:38:38.000000000 +0000
+++ mozilla/js/src/configure.in
@@ -1200,7 +1200,6 @@ if test "$GNU_CC"; then
 
     # Treat some warnings as errors:
     _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Werror=address"
-    _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Werror=char-subscripts"
     _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Werror=comment"
     _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Werror=empty-body"
     _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Werror=endif-labels"
@@ -2180,8 +2179,7 @@ AC_LANG_CPLUSPLUS
 
 MOZ_CXX11
 
-dnl Check for .hidden assembler directive and visibility attribute.
-dnl Borrowed from glibc configure.in
+dnl Setup default hidden visibility and wrapped system headers.
 dnl ===============================================================
 if test "$GNU_CC" -a "$OS_TARGET" != WINNT; then
   AC_DEFINE(HAVE_VISIBILITY_HIDDEN_ATTRIBUTE)
@@ -3095,7 +3093,7 @@ if test "$MOZ_MEMORY"; then
   *-darwin*)
     AC_DEFINE(MOZ_MEMORY_DARWIN)
     ;;
-  *-*freebsd*)
+  *-*freebsd*|*-*dragonfly*)
     AC_DEFINE(MOZ_MEMORY_BSD)
     ;;
   *-android*|*-linuxandroid*)
@@ -3989,6 +3987,16 @@ MOZ_SUBCONFIGURE_ICU()
 dnl ========================================================
 dnl JavaScript shell
 dnl ========================================================
+ICU_LIB_NAMES=
+MOZ_NATIVE_ICU=
+MOZ_ARG_WITH_BOOL(system-icu,
+[  --with-system-icu
+                          Use system icu (located with pkgconfig)],
+    MOZ_NATIVE_ICU=1)
+
+if test -n "$MOZ_NATIVE_ICU"; then
+    PKG_CHECK_MODULES(MOZ_ICU, icu-i18n >= 50.1)
+fi
 
 MALLOC_HEADERS="malloc.h malloc_np.h malloc/malloc.h sys/malloc.h"
 MALLOC_H=
